src/Components/ToggleSave/ToggleSave.js

```diff
@@ -1,35 +1,60 @@
-import React, { useState } from "react";
+
+import React from "react";
 import toggleOffImage from "../images/image_png.png";
 import toggleOnImage from "../images/color toggle (1).png";
 import "../ToggleSave/ToggleSave.css";
-import { useMutation } from "@apollo/client";
-import { TOGGLE_SAVE_MUSHROOM } from "../../queries";
-
-const ToggleSave = ({ mushroomId, isSavedInitially, onToggle }) => {
-  const [isToggled, setIsToggled] = useState(isSavedInitially);
-  const [toggleSaveMushroom] = useMutation(TOGGLE_SAVE_MUSHROOM);
-  const userId = 3;
-
-  const handleToggle = async () => {
-    try {
-      console.log("User ID:", userId); 
-      console.log("Mushroom ID:", mushroomId);
-      
-      await toggleSaveMushroom({
-        variables: { userId, mushroomId },
+import { useMutation, useQuery } from "@apollo/client";
```

Comment:
now using both mutation and query functionalities from Apollo Client.!

really cool